Exercise_3.java

```diff
@@ -13,13 +17,29 @@
             next = null; 
         } 
     } 
-  
+
     /* Function to print middle of linked list */
    //Complete this function
-    void printMiddle() 
+  
+  
+  
+  
+    void printMiddle()  
     { 
         //Write your code here
 	//Implement using Fast and slow pointers
+        if((head.next == null) || (head.next.next == null)){     //For size 1 or 2, im assuming that 1st node will be middle.
+            System.out.println(head.data);
+        }
+        else{
+            Node slow = head.next;      
+            Node fast = head.next.next;
+            while((fast.next != null) && (fast.next.next != null)){ //when fast pointer meets the end of list, slow pointer will be at middle
+               slow = slow.next;
+               fast = fast.next.next;
+            }
+            System.out.println(slow.data);
+        }     
     } 

```

Comment:
a better way to handle all the cases without if else block would be:
        Node slow = head;
        Node fast = head;
        while (fast.next != null && fast.next.next != null) {
            slow = slow.next;
            fast = fast.next.next;
        }
        System.out.print(slow.data + " ");